src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -0,0 +1,418 @@
+                expectedDefault: true,
+                expectedIncludeNullability: true
+                );
+
+            VerifyEquality(type1comp2, type2,
+                expectedDefault: true,
+                expectedIncludeNullability: true
+                );
+        }
+
+        [Fact]
+        public void SemanticModel_Symbol_Equality()
+        {
+            var source =
+@"
+public class A
+{
+    public static A field1;
+    public static A field2;
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var member1Syntax = (FieldDeclarationSyntax)root.DescendantNodes().First(sn => sn.Kind() == SyntaxKind.FieldDeclaration);
+            var member2Syntax = (FieldDeclarationSyntax)root.DescendantNodes().Last(sn => sn.Kind() == SyntaxKind.FieldDeclaration);
+
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var member1 = model.GetDeclaredSymbol(member1Syntax.Declaration.Variables[0]);
+            var member2 = model.GetDeclaredSymbol(member2Syntax.Declaration.Variables[0]);
+
+            Assert.Equal(true, member1.Equals(member1));
+            Assert.Equal(true, member2.Equals(member2));
+            Assert.Equal(false, member1.Equals(member2));
+            Assert.Equal(false, member2.Equals(member1));
+
+            var field1 = member1 as FieldSymbol;
+            var field2 = member2 as FieldSymbol;
+
+            Assert.Equal(true, field1.Equals(field1));
+            Assert.Equal(true, field2.Equals(field2));
+            Assert.Equal(false, field1.Equals(field2));
+            Assert.Equal(false, field2.Equals(field1));
+        }
+
+        [Fact]
+        public void SemanticModel_Type_Equality_With_No_Nullability()
+        {
+            var source =
+@"
+public class A
+{
+    public static A field1;
+    public static A field2;
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var member1Syntax = (FieldDeclarationSyntax)root.DescendantNodes().First(sn => sn.Kind() == SyntaxKind.FieldDeclaration);
+            var member2Syntax = (FieldDeclarationSyntax)root.DescendantNodes().Last(sn => sn.Kind() == SyntaxKind.FieldDeclaration);
+
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var type1 = (model.GetDeclaredSymbol(member1Syntax.Declaration.Variables[0]) as IFieldSymbol).Type;
+            var type2 = (model.GetDeclaredSymbol(member2Syntax.Declaration.Variables[0]) as IFieldSymbol).Type;
+
+            VerifyEquality(type1, type2,
+                expectedDefault: true,
+                expectedIncludeNullability: true
+            );
+        }
+
+        [Fact]
+        public void SemanticModel_Type_Equality_With_Top_Level_Nullability()
```

Comment:
Please add a test on the IFieldSymbol itself, verifying that we do consider top-level nullability in that case.

Reply:
How can you have a field symbol that is the same except for nullability? (without generics)

---
In reply to: [305127618](https://github.com/dotnet/roslyn/pull/37247#discussion_r305127618) [](ancestors = 305127618)

Reply:
You can have a field on a substituted generic type, but I don't think that counts as having different top-level nullability, that would be different nested nullability (I think?)